lib/infra/log/log.go

```diff
@@ -0,0 +1,121 @@
+package log
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// Level indicates the severity of the data being logged
+type Level string
+
+var (
+	// LevelDebug are debug or trace information
+	LevelDebug Level = "DEBUG"
+	// LevelInfo are routine information
+	LevelInfo Level = "INFO"
+	// LevelWarning warns about events the might cause problems to the system
+	LevelWarning Level = "WARNING"
+	// LevelError alerts about events that are likely to cause problems
+	LevelError Level = "ERROR"
+	// LevelCritical alerts about severe problems. Most of the time, needs some human intervention ASAP
+	LevelCritical Level = "CRITICAL"
+)
+
+var levelPriority = map[Level]int{
+	LevelCritical: 5,
+	LevelError:    10,
+	LevelWarning:  15,
+	LevelInfo:     20,
+	LevelDebug:    25,
+}
```

Comment:
I got this idea to remove repeated literals from the code:
```suggestion
// Level indicates the severity of the data being logged
type Level int

const (
	// LevelDebug are debug or trace information
	LevelDebug Level = iota
	// LevelInfo are routine information
	LevelInfo
	// LevelWarning warns about events the might cause problems to the system
	LevelWarning
	// LevelError alerts about events that are likely to cause problems
	LevelError
	// LevelCritical alerts about severe problems. Most of the time, needs some human intervention ASAP
	LevelCritical
)

func (l Level) String() string {
	return []string{
		"DEBUG",
		"INFO",
		"WARNING",
		"ERROR",
		"CRITICAL",
	}[l]
}
```

Reply:
@RaiTamarindo nice call!!